src/app/modules/shared_models/message_item.nim

```diff
@@ -38,6 +38,7 @@ type
     mentionedUsersPks: seq[string]
     senderTrustStatus: TrustStatus
     senderEnsVerified: bool
+    discordMessage: DiscordMessageItem
```

Comment:
I like that idea. I've tried to stick to existing patterns (notice how `message_item.nim`' `Item` also takes a `TransactionsParametersItem` as view model object.) And if you check below, we're actually already using most of the `discordMessage` properties to hydrate `Item`.

I'll try to not be reliant of `discordMessage` as a view property in QML, but I can't promise yet that this will work for all cases.

Already starts with the fact that a `discordMessage` can have many `attachments` (images, files etc).
`MessageItem` only supports a single image.

There might be more things

Reply:
BTW: this is the place where we make use of `discordMessage`'s data: https://github.com/status-im/status-desktop/pull/7095/files#diff-4f86d004be69cc9b51a70273a5ec7ab80e851a4231773dad079e8c01afa753b5R103-R112

We could also move this logic out of the `newItem` constructor but then we'll have to do it for every single `newItem` call for message_Item.

Reply:
So in other words, I'd say: It's fine to **at least** pass the `DiscordMesage` model to `MessageItem` in the constructor so we can access its data if needed. But then not necessarily make it a `DiscordMessageItem` and ideally not rely on `message.discordMessage` in QML (as mentioned, to be figured out)

Reply:
> Ideally, the client (qml code) shouldn't know the message is coming from Discord. For example, we could have message.Item.label == "Imported from Discord" for Discord messages and message.Item.label == "" for ordinary Status messages.

Giving this further thought, I think I tend to disagree with this part. We shouldn't put any "hard coded" labels into our application code but rather have the view figure it out. "Imported from discord" is a label that needs to be i18n'd as well. So better to have some type or value to check against and then decide what kind of label to show inside of QML

Reply:
status-go already provide message IDs to link multiple image messages, allowing clients (like desktop) to render them as single message.

I'll have to look into what that entails and then figure out if we can make it work similarly for discord message provided content.